rust-runtime/aws-smithy-runtime-api/src/client/http/request.rs

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::box_error::BoxError;
+use aws_smithy_http::body::SdkBody;
+use http as http0;
+use http0::header::Iter;
+use http0::{Extensions, HeaderMap, Method};
+use std::borrow::Cow;
+use std::fmt::Debug;
+use std::str::FromStr;
+
+#[derive(Debug)]
+/// An HTTP Request Type
+pub struct Request<B = SdkBody> {
```

Comment:
Have you tried using the Request/Response with Surf? Does it play nicely with that API?

Reply:
the API works fine _except_ for extensions. We'd need to maintain a copy of each extension map we aim to target or the maps would need to support construction from TypeId directly